### PR TITLE
cgroup-limit: support sudo for using cpu limits, and running as root

### PR DESCRIPTION
**Pull request by tmds**
Fixes https://github.com/redhat-developer/dotnet-regular-tests/issues/139

### CODE REVIEW

**Comment by tmds** [comment]
@omajid I think this will solve most/all issues we see. When you had a look, we can merge it and see the effect in our CI.

File: cgroup-limit/test.sh [hunk 1]
```diff
@@ -6,7 +6,18 @@ set -x
 
 dotnet publish
 
-mapfile -t DOTNET_LIMITS < <(systemd-run -q --scope --user -p CPUQuota=100% -p MemoryLimit=100M bin/Debug/net*/cgroup-limit)
+SYSTEMD_RUN="systemd-run"
+if [ "$UID" != "0" ]; then
+  if ! grep -q "cpu" "/sys/fs/cgroup/user.slice/user-$UID.slice/user@$UID.service/cgroup.controllers" ; then
+    # user can't set cpu limits, use sudo.
+    SYSTEMD_RUN="sudo -n $SYSTEMD_RUN"
```
Comment (omajid) [inline cgroup-limit/test.sh:13]: :+1: for `-n`

**Review by omajid — APPROVED** [review]

**Comment by tmds** [comment]
Thanks for the review @omajid !

**Comment by omajid** [comment]
Thanks for the fix! This passes in Fedora CI now: https://osci-jenkins-1.ci.fedoraproject.org/job/fedora-ci/job/dist-git-pipeline/job/master/35223/testReport/(root)/tests/regular/

**Pull request merged**